source/slang/slang-target-program.cpp

```
@@ -59,29 +69,37 @@ IArtifact* TargetProgram::_createEntryPointResult(
     {
-        m_entryPointResults[entryPointIndex].setNull();
         return nullptr;
     }
 
+    {
+        std::lock_guard<std::mutex> lock(m_resultCacheMutex);
+        if (entryPointIndex >= m_entryPointResults.getCount())
+            m_entryPointResults.setCount(entryPointIndex + 1);
+        if (m_entryPointResults[entryPointIndex])
+            return m_entryPointResults[entryPointIndex];
+        m_entryPointResults[entryPointIndex] = artifact;
```

Cppcheck (2.20.0)

[information] Include file

(missingInclude)